tuist/tuist
# Support optional frameworks and xcframeworks

Resolves https://github.com/tuist/tuist/issues/5520

### Short description 📝

This change adds support for defining optional `.framework` or `.xcframework` dependencies to targets. This change adds a `status` field to the `TargetDependency.framework` and `TargetDependency.xcframework` cases. `status` is an enum value that defaults to `.required` for backwards compatibility. If `status` is set to `.optional`, then when the reference is written to the `.pbxproj` file, the `ATTRIBUTES` setting will be set to `Weak`.

### How to test the changes locally 🧐

1. Create a Tuist project.
2. Add a `.framework` or `.xcframework` to the project file system.
3. When adding the framework to a target, specify `required: false` to the `TargetDependency` value:

```swift
let project = Project(
    ...
    targets: [
        .init(
            ...
            dependencies: [
               ...
                .xcframework(path: "path-to-xcframework", status: .optional),
                ...
            ]
        )
    ],
    ...
)
```

4. Run `tuist generate` to generate the workspace and project files.
5. Open the workspace in Xcode and navigate to the target settings. Go to the __Build Phases__ tab and expand the __Link Binary With Libraries__ phase. The __Status__ field for the framework should show `Optional` instead of `Required`.

### Contributor checklist ✅


### Reviewer checklist ✅
